### PR TITLE
Add and fix compiler warnings.

### PR DESCRIPTION
**Pull request by jkinkead**
There were a number of unchecked operations (especially on `RendezvousHash` usage), and a few calls to deprecated methods.

`addCallback` in particular is slated to be removed in April.

### CODE REVIEW

File: src/main/java/com/nordstrom/xrpc/client/Call.java [hunk 1]
```diff
@@ -91,7 +92,9 @@ public void onFailure(Throwable t) {
             response.cancel(true);
             error.setException(t);
           }
-        });
+        },
+        MoreExecutors.directExecutor()
```
Comment (andyday) [inline src/main/java/com/nordstrom/xrpc/client/Call.java]: can you explain this usage?  not that obvious...

File: src/test/java/com/nordstrom/xrpc/client/XrpcClientTest.java [hunk 2]
```diff
@@ -63,7 +64,9 @@ public void onFailure(Throwable t) {
             latch.countDown();
             assertEquals(true, false);
           }
-        });
+        },
+        MoreExecutors.directExecutor()
```
Comment (andyday) [inline src/test/java/com/nordstrom/xrpc/client/XrpcClientTest.java]: see comment above...

File: src/main/java/com/nordstrom/xrpc/client/Call.java (hunk 1 above)
Comment (jkinkead) [reply src/main/java/com/nordstrom/xrpc/client/Call.java]: [See the deprecation note](http://google.github.io/guava/releases/snapshot-jre/api/docs/com/google/common/util/concurrent/Futures.html#addCallback-com.google.common.util.concurrent.ListenableFuture-com.google.common.util.concurrent.FutureCallback-). This is replacing what was an implicit executor with the explicit version.

The concept of a direct (threadless) executor is used all over when operating within the java.util.concurrent world.

**Comment by andyday** [comment]
formatting error -> travis

**Pull request merged**